chapters/en/Unit 2 - Convolutional Neural Networks/googlenet.mdx

```diff
@@ -4,44 +4,101 @@
+
+We would also want to reduce the output features of max pooling before concatenating with the output feature map. So, we add one more $1\times 1$ convolution after the maxpooling layer. We also add a ReLU activation after each $1\times 1$ convolution increasing non linearity and complexity of the module. See Figure 2.
+
+![](https://huggingface.co/datasets/hf-vision/course-assets/resolve/main/inception_reduced.png)
+Figure 2
+
+Also, because of the parallel operations of convolutions at multiple scales, we are ensuring more operations without going deeper into the network, essentially mitigating the vanishing gradient problem.
+
+#### 2.2 Average Pooling
+
+In prior networks, like AlexNet or VGG, the final layers would be a few fully connected layers. These fully connected layers, due to their large number of units would contribute to most of the parameters in a network. For example, 89% of the parameters of VGG16 are in the final three fully connected layers. 95% of parameters in AlexNet are in the final fully connected layers. This need can be attributed to the premise that a Convolutional layer is not necessarily complex enough. 
+
+However, with an Inception block at our disposal, we do not need fully connected layers and a simple average pooling along the spatial dimensions should be enough. This was also derived from the Network in Network paper. However, GoogLeNet included one fully connected layer. They reported an increase of 0.6% in top-1 accuracy. 
+
+GoogLeNet has only 15% of the parameters in the fully connected layers.
+
+
+#### 2.3 Auxiliary Classifiers
+
+With the introduction of compute saving $1 \times 1$ convolution and the replacement of multiple fully connected layers with average pooling, the parameters of this network are reduced significantly, which means we can add more layers and go deeper into the network. However, stacking layers can cause the problem of "vanishing gradient", where the gradients get smaller and close to zero while propagating back to the initial layers of the network.
+
+The paper introduces the auxiliary classifiers - branch out a few small classifiers from the layers in between and add the loss from these classifiers to the total loss(with less weightage). This ensures that the layers close to the input also receive gradients of decent magnitude.
+
+The auxiliary classifier consists of 
+- An average pooling layer with $5 \times 5$ filter size and stride 3.
+- A $1 \times 1$ convolution with 128 filters for dimension reduction and rectified linear activation.
+- A fully connected layer with 1024 units and rectified linear activation.
+- A dropout layer with 70% ratio of dropped outputs.
+- A linear layer with softmax loss as the classifier.
+
+These auxiliary classifiers are removed at inference time. However, minimal gains are achieved from using auxiliary classifiers (0.5%).
+
+<!-- <<<maybe insert figure>> -->
```

Comment:
@themurtazanazir are you sure which figure to add?

Reply:
Yes I do.

Reply:
@themurtazanazir can you give the diagram to @sitamgithub-MSIT so that you can add the link here?

Reply:
Done already

Reply:
the figures are added.